src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/JsonValueConverterKeyValuePair.cs

```diff
@@ -3,22 +3,26 @@
 // See the LICENSE file in the project root for more information.
 
 using System.Collections.Generic;
-using System.Diagnostics.CodeAnalysis;
 
 namespace System.Text.Json.Serialization.Converters
 {
-    internal sealed class JsonKeyValuePairConverter<TKey, TValue> : JsonConverter<KeyValuePair<TKey, TValue>>
+    internal sealed class JsonKeyValuePairConverter<TKey, TValue> : JsonValueConverter<KeyValuePair<TKey, TValue>>
     {
         private const string KeyName = "Key";
         private const string ValueName = "Value";
 
-        // "encoder: null" is used since the literal values of "Key" and "Value" should not normally be escaped
-        // unless a custom encoder is used that escapes these ASCII characters (rare).
-        // Also by not specifying an encoder allows the values to be cached statically here.
+        // todo: move these to JsonSerializerOptions and use the proper encoding.
         private static readonly JsonEncodedText _keyName = JsonEncodedText.Encode(KeyName, encoder: null);
         private static readonly JsonEncodedText _valueName = JsonEncodedText.Encode(ValueName, encoder: null);
 
-        public override KeyValuePair<TKey, TValue> Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
+        // todo: it is possible to cache the underlying converters since this is an internal converter and
+        // an instance is created only once for each JsonSerializerOptions instance.
```

Comment:
Do have tracking issues for these?

Reply:
No, but the issue existed before the PR. I did not add the caching code here (yet) since currently this is the only code path that calls the new re-entry JsonSerializer method. I'll add an issue once this PR is in.